homeassistant/components/binary_sensor/insteon_plm.py

```diff
@@ -2,86 +2,66 @@
 Support for INSTEON dimmers via PowerLinc Modem.
 
 For more details about this component, please refer to the documentation at
-https://home-assistant.io/components/insteon_plm/
+https://home-assistant.io/components/binary_sensor.insteon_plm/
 """
-import logging
 import asyncio
+import logging
 
-from homeassistant.core import callback
+from homeassistant.components.insteon_plm import InsteonPLMEntity
 from homeassistant.components.binary_sensor import BinarySensorDevice
-from homeassistant.loader import get_component
+from homeassistant.core import callback
 
 DEPENDENCIES = ['insteon_plm']
 
 _LOGGER = logging.getLogger(__name__)
 
+SENSOR_TYPES = {'openClosedSensor': 'opening',
+                'motionSensor': 'motion',
+                'doorSensor': 'door',
+                'leakSensor': 'moisture'}
+
 
 @asyncio.coroutine
 def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up the INSTEON PLM device class for the hass platform."""
     plm = hass.data['insteon_plm']
 
-    device_list = []
-    for device in discovery_info:
-        name = device.get('address')
-        address = device.get('address_hex')
+    address = discovery_info['address']
+    device = plm.devices[address]
+    state_key = discovery_info['state_key']
 
-        _LOGGER.info('Registered %s with binary_sensor platform.', name)
+    _LOGGER.debug(
+        'Adding device %s with state name %s to Binary Sensor platform.',
+        device.address.hex, device.states[state_key].name)
 
-        device_list.append(
-            InsteonPLMBinarySensorDevice(hass, plm, address, name)
-        )
+    new_entity = InsteonPLMBinarySensor(device, state_key)
 
-    async_add_devices(device_list)
+    async_add_devices([new_entity])
 
 
-class InsteonPLMBinarySensorDevice(BinarySensorDevice):
-    """A Class for an Insteon device."""
-
-    def __init__(self, hass, plm, address, name):
-        """Initialize the binarysensor."""
-        self._hass = hass
-        self._plm = plm.protocol
-        self._address = address
-        self._name = name
-
-        self._plm.add_update_callback(
-            self.async_binarysensor_update, {'address': self._address})
-
-    @property
-    def should_poll(self):
-        """No polling needed."""
-        return False
+class InsteonPLMBinarySensor(InsteonPLMEntity, BinarySensorDevice):
+    """A Class for an Insteon device entity."""
 
-    @property
-    def address(self):
-        """Return the address of the node."""
-        return self._address
+    @callback
+    def async_binarysensor_update(self, deviceid, statename, val):
```

Comment:
You can move this method to the common base class too. Just rename it to something general like `update_callback`.